godot-core/src/builtin/meta/signature.rs

```diff
@@ -35,6 +35,14 @@ pub trait SignatureTuple {
         method_name: &str,
     );
 
+    unsafe fn varcall_static<C: GodotClass>(
+        args_ptr: *const sys::GDExtensionConstVariantPtr,
+        ret: sys::GDExtensionVariantPtr,
+        err: *mut sys::GDExtensionCallError,
+        func: fn(Self::Params) -> Self::Ret,
+        method_name: &str,
+    );
```

Comment:
In #210, I made the `func` accept `*const sys::GDExtensionConstVariantPtr`. This pushes the responsibility of converting from `GDExtensionConstVariantPtr` to `&C`, `&mut C` or nothing at all up to the caller, but allows us to have one `varcall` and one `ptrcall` implementation, instead of one for each of the ref, mut and static cases. That cuts down on duplication in this file.

Since you already have a macro that takes "magic words" `ref`, `mut` or `static` it shouldn't be hard to do the same in this PR as well, if we want to.

Reply:
it did help a lot with code repetition tbh, so i changed to doing that